itou/openid_connect/inclusion_connect/views.py

```diff
@@ -74,34 +92,36 @@ def inclusion_connect_authorize(request):
     request = ic_session.bind_to_request(request)
     ic_session = request.session[constants.INCLUSION_CONNECT_SESSION_KEY]
 
-    redirect_uri = get_absolute_url(reverse("inclusion_connect:callback"))
-    signed_csrf = InclusionConnectState.create_signed_csrf_token()
-    data = {
-        "response_type": "code",
-        "client_id": constants.INCLUSION_CONNECT_CLIENT_ID,
-        "redirect_uri": redirect_uri,
-        "scope": constants.INCLUSION_CONNECT_SCOPES,
-        "state": signed_csrf,
-        "nonce": crypto.get_random_string(length=12),
-        "from": "emplois",  # Display a "Les emplois" logo on the connection page.
-    }
     login_hint = request.GET.get("login_hint")
     channel = request.GET.get("channel")
     if login_hint:
         if not channel:
             return _redirect_to_login_page_on_error(error_msg="channel is missing when login_hint is present.")
-        data["login_hint"] = login_hint
         ic_session["user_email"] = login_hint
         ic_session["channel"] = channel
         request.session.modified = True
 
+    data = generate_inclusion_params_from_session(ic_session)
+
     if request.GET.get("register"):
         base_url = constants.INCLUSION_CONNECT_ENDPOINT_REGISTER
     else:
         base_url = constants.INCLUSION_CONNECT_ENDPOINT_AUTHORIZE
     return HttpResponseRedirect(f"{base_url}?{urlencode(data)}")
 
 
+def inclusion_connect_resume_registration(request):
+    """
+    Used for users that didn't received the validation e-mail from Inclusion Connect.
+    This view will allow them to resume the account creation flow.
+    """
+    ic_session = request.session.get(constants.INCLUSION_CONNECT_SESSION_KEY)
+    if not ic_session or ic_session["token"]:
```

Comment:
```suggestion
    if not (ic_session or ic_session["token"]):
```

C'est plus explicite, je trouve.
J'ai eu du mal à comprendre à la première lecture car je me suis demandée si la clé token était toujours présente (oui, elle l'est). Bref.

Reply:
Ça ne marche pas :
ma version c'est soit pas de `ic_session`, soit `ic_session` existe et `ic_session["token"]` aussi.
Dans ta version, si on a `ic_session` avec un `token` dedans, alors `ic_session or ic_session["token"]` est à True, et `not (ic_session or ic_session["token"])` à False.
Je vais mettre un commentaire pour que ce soit plus explicite.

Reply:
Ça marche ! :)